src/main/java/de/tum/in/www1/artemis/service/LectureUnitProcessingService.java

```diff
@@ -0,0 +1,158 @@
+            AttachmentUnit attachmentUnit = new AttachmentUnit();
+            Attachment attachment = new Attachment();
+            PDDocumentInformation pdDocumentInformation = new PDDocumentInformation();
+
+            pdfSplitter.setStartPage(lectureUnit.startPage());
+            pdfSplitter.setEndPage(lectureUnit.endPage());
+            pdfSplitter.setSplitAtPage(lectureUnit.endPage());
+
+            List<PDDocument> documentUnits = pdfSplitter.split(document);
+            pdDocumentInformation.setTitle(lectureUnit.unitName());
+            documentUnits.get(0).setDocumentInformation(pdDocumentInformation);
+            documentUnits.get(0).save(outputStream);
+
+            // setup attachmentUnit and attachment
+            attachmentUnit.setDescription("");
+            attachment.setName(lectureUnit.unitName());
+            attachment.setAttachmentType(AttachmentType.FILE);
+            attachment.setVersion(1);
+            attachment.setReleaseDate(lectureUnit.releaseDate());
+            attachment.setUploadDate(ZonedDateTime.now());
+
+            // prepare file to be set from byte[] to MultipartFile by using CommonsMultipartFile
+            String tempDirectory = System.getProperty("java.io.tmpdir");
+            File tempFile = new File(tempDirectory, lectureUnit.unitName() + ".pdf");
+            Files.write(tempFile.toPath(), outputStream.toByteArray());
+            File outputFile = new File(tempFile.toString());
+            FileItem fileItem = new DiskFileItem("mainUnitFile", Files.probeContentType(outputFile.toPath()), false, outputFile.getName(), (int) outputFile.length(),
+                    outputFile.getParentFile());
+            InputStream input = new FileInputStream(outputFile);
+            OutputStream fileItemOutputStream = fileItem.getOutputStream();
+            IOUtils.copy(input, fileItemOutputStream);
+            MultipartFile multipartFile = new CommonsMultipartFile(fileItem);
+
+            LectureUnitDTO lectureUnitsDTO = new LectureUnitDTO(attachmentUnit, attachment, multipartFile);
+            units.add(lectureUnitsDTO);
+            tempFile.deleteOnExit();
+        }
+
+        document.close();
+        return units;
+    }
+
+    /**
+     * Prepare information of split units for client
+     * @param file The file (lecture slide) to be split
+     * @return The prepared information of split units as list of LectureUnitSplitDTO
+     */
+    public LectureUnitInformationDTO getSplitUnitData(MultipartFile file) throws IOException {
+
+        List<LectureUnitSplitDTO> units = new ArrayList<>();
+        PDDocument document = PDDocument.load(file.getBytes());
+        Tuple<Map<Integer, Tuple<String, Tuple<Integer, Integer>>>, Integer> unitsInformation = separateIntoUnits(document);
+        Map<Integer, Tuple<String, Tuple<Integer, Integer>>> unitsDocumentMap = unitsInformation.x();
+        int numberOfPages = unitsInformation.y();
+
+        unitsDocumentMap.forEach((k, v) -> {
```

Comment:
Avoid short variable names like `k` and `v`. Below you use `v.y().x(), v.y().y()` which makes it impossible to understand what is going on by reading the code. (Should be solved by introducing the `Outline` type)

Reply:
now should look better :+1: